johnzon-jsonb/src/test/java/org/apache/johnzon/jsonb/JsonbWriteTest.java

```diff
@@ -92,6 +94,14 @@ public void list() {
         assertEquals("[\"a\",\"b\"]", JsonbProvider.provider().create().build().toJson(map));
     }
 
+    @Test
+    public void listOfSimple() {
```

Comment:
@rmannibucau Here is the test, consistently fails (for me) on the master (unpatched) on JDK16-ea+28 and above:
```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.johnzon.jsonb.JsonbWriteTest
[ERROR] Tests run: 12, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.27 s <<< FAILURE! - in org.apache.johnzon.jsonb.JsonbWriteTest
[ERROR] listOfSimple(org.apache.johnzon.jsonb.JsonbWriteTest)  Time elapsed: 0.016 s  <<< ERROR!
java.lang.reflect.InaccessibleObjectException: Unable to make field private int java.util.ArrayList.size accessible: module java.base does not "opens java.util" to unnamed module @6b71769e
        at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:357)
        at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
        at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:177)
        at java.base/java.lang.reflect.Field.setAccessible(Field.java:171)
        at org.apache.johnzon.mapper.access.FieldAccessMode$FieldDecoratedType.<init>(FieldAccessMode.java:105)
        at org.apache.johnzon.mapper.access.FieldAccessMode$FieldReader.<init>(FieldAccessMode.java:169)
        at org.apache.johnzon.mapper.access.FieldAccessMode.doFindReaders(FieldAccessMode.java:49)
        at org.apache.johnzon.mapper.access.BaseAccessMode.findReaders(BaseAccessMode.java:82)
        at org.apache.johnzon.mapper.access.FieldAndMethodAccessMode.doFindReaders(FieldAndMethodAccessMode.java:68)
        at org.apache.johnzon.mapper.access.BaseAccessMode.findReaders(BaseAccessMode.java:82)
        at org.apache.johnzon.jsonb.JsonbAccessMode.findReaders(JsonbAccessMode.java:468)
        at org.apache.johnzon.mapper.Mappings.createClassMapping(Mappings.java:475)
        at org.apache.johnzon.mapper.Mappings.doFindOrCreateClassMapping(Mappings.java:437)
        at org.apache.johnzon.mapper.Mappings.findOrCreateClassMapping(Mappings.java:411)
        at org.apache.johnzon.mapper.Mapper.isDeduplicateObjects(Mapper.java:215)
        at org.apache.johnzon.mapper.Mapper.writeObjectWithGenerator(Mapper.java:209)
        at org.apache.johnzon.mapper.Mapper.writeObject(Mapper.java:203)
        at org.apache.johnzon.mapper.Mapper.writeObjectAsString(Mapper.java:252)
        at org.apache.johnzon.jsonb.JohnzonJsonb.toJson(JohnzonJsonb.java:342)
        at org.apache.johnzon.jsonb.JsonbWriteTest.listOfSimple(JsonbWriteTest.java:102)
```

Reply:
Interesting, just read on the phone but our list detection is something along "is parameterized and raw type is a collection or is an array" so passing List.class was moving it to a fallback handling and lead to the unexpected stack you get.
Guess it is what we must fix and probably for maps too.

Hope it makes sense

Reply:
Yeah, in this particular case for ArrayList it could be certainly done, I run into it by accindent  ;-). I am wondering if you were able to see the other failures, `ReadPrimitiveTest` & `CircularExceptionTest`?

Reply:
Didnt run it but looks like it is the same issue of "too late type handling" so guess fix is to move it earlier trying to limit runtime impact.
If not urgent can work on it next week.

Reply:
@rmannibucau no, it is not urgent, please let me know if I could help in scope of this PR (or I can just close it).

Reply:
Let's keep it open until we review the final solution and agree on it. Please ping me if you dont get news in 10 days ;)

Reply:
Hey @rmannibucau, I gave it a second thought, particularly when you mentioned the presence of the extensions and completely refactored the approach to eliminate flows driven by exceptions. I think it is more clean but let us discuss it when you have time, thanks.

Reply:
Hi @reta,

I like this version more but it still fixes the consequence instead of the source of the issue which is an unexpected branch in the mapper (class for collection/primitive/map/throwable/jdk-class) so even if this flavor can be kept I still think we should fix first the cause then maybe review this flavor.

Wdyt?

Reply:
@rmannibucau This very correct, I will continue digging in so we could be proactive in introspection (we are indeed reactive now), will be back on that :) Thanks for the feedback.